FluentTerminal.SystemTray/Services/TerminalsManager.cs

```diff
@@ -8,36 +9,98 @@
 using System.Collections;
 using System.Collections.Generic;
 using System.Diagnostics;
+using System.IO;
 using System.Text;
+using System.Text.RegularExpressions;
 using Windows.ApplicationModel;
 
 namespace FluentTerminal.SystemTray.Services
 {
+    public struct TerminalSessionInfo
+    {
+        public DateTime StartTime { get; set; }
+        public string ProfileName { get; set; }
+        public ITerminalSession Session { get; set; }
+    }
+
     public class TerminalsManager
     {
-        private readonly Dictionary<byte, ITerminalSession> _terminals = new Dictionary<byte, ITerminalSession>();
+        private readonly Dictionary<byte, TerminalSessionInfo> _terminals = new Dictionary<byte, TerminalSessionInfo>();
 
         public event EventHandler<TerminalOutput> DisplayOutputRequested;
 
         public event EventHandler<TerminalExitStatus> TerminalExited;
 
+        private readonly ISettingsService _settingsService;
+
+        private static readonly Regex EscapeSequencePattern = new Regex(@"((\x9B|\x1B\[)[0-?]*[ -\/]*[@-~])|((\x9D|\x1B\]).*\x07)", RegexOptions.Compiled);
+
+        private Dictionary<byte, string> _cachedLogPath = new Dictionary<byte, string>();
+
+        public TerminalsManager(ISettingsService settingsService)
+        {
+            _settingsService = settingsService;
+        }
+
         public void DisplayTerminalOutput(byte terminalId, byte[] output)
         {
+            var appSettings = _settingsService.GetApplicationSettings();

```

Comment:
still not happy with this. We should keep this as a field and update it on change (I think the UWP is already informing the SystemTray when ApplicationSettings have changed)

Reply:
Unfortunately, there is no special Request class to inform SystemTray about changed settings or can't see any other notification mechanism. Going to add SettingsChangedRequest to be sent from UWP to Tray process.

Reply:
@felixse here we go https://github.com/felixse/FluentTerminal/pull/438/commits/49b5d4ef97295282f89b98925779b09d1461b72e